marklogic-data-hub/src/main/java/com/marklogic/hub/impl/HubProjectImpl.java

```diff
@@ -769,6 +769,43 @@ public Path getStepsPath() {
         return this.projectDir.resolve("steps");
     }
 
+    @Override
+    public Path getStepsPath(StepDefinition.StepDefinitionType type) {
+        Path path;
+
+        Path parent = this.getStepsPath();
+
+        if (type == null) {
+            throw new DataHubProjectException("Invalid Step type");
+        }
+        else {
+            switch (type) {
+                case CUSTOM:
+                    path = parent.resolve("custom");
+                    break;
+                case INGESTION:
+                    path = parent.resolve("ingestion");
+                    break;
+                case MAPPING:
+                    path = parent.resolve("mapping");
+                    break;
+                case MASTERING:
+                    path = parent.resolve("mastering");
+                    break;
+                case MATCHING:
+                    path = parent.resolve("matching");
+                    break;
+                case MERGING:
+                    path = parent.resolve("merging");
+                    break;
+                default:
+                    throw new DataHubProjectException("Invalid Step type");
```

Comment:
Can you include the value of type in here - e.g. "Invalid step type: " + type

Reply:
Done